File: demos/jans-chip/kmp/iosApp/SuperGluuFido/DI/nativeModule.swift
//  SuperGluuFido
//
//  Created by Nazar Yavornytskyi on 01.10.2024.
//  Copyright © 2024 orgName. All rights reserved.
//

import Foundation
import shared

var nativeModule: Koin_coreModule = MakeNativeModuleKt.makeNativeModule(
    analytic: { scope in
        return AnalyticImpl(logger: scope.get())
    },
    authentication: { _ in
        return AuthenticationAdaptor()
    },
    dPoPProofProvider: { _ in
        return DPoPProofFactoryAdaptor()
    }
)

File: demos/jans-chip/kmp/composeApp/src/androidMain/kotlin/di/nativeModule.kt
package di

import com.example.fido2.di.makeNativeModule
import services.AnalyticImpl
import services.AuthAdaptor
import services.DPoPProofFactoryAdaptor

val nativeModule = makeNativeModule(
    analytic = { AnalyticImpl( get() ) },
    authentication = { AuthAdaptor(get()) },
    dPoPProofProvider = { DPoPProofFactoryAdaptor(get()) }
)